plugins/database/src/db_plugin.cpp

```diff
@@ -694,6 +695,43 @@ irods::error validate_resource_name( std::string _resc_name ) {
 
 } // validate_resource_name
 
+// Not super robust
+static inline auto validate_zone_connection_string(const char* _zone_conn_info, irods::plugin_context& _ctx)
+    -> irods::error
+{
+    const std::size_t addr_len = std::strlen(_zone_conn_info);
+    if (addr_len > 0) {
+        // _zone_conn_info is const, so copy it
+        std::vector<char> addr_buf(addr_len + 1, '\0');
+        std::strncpy(addr_buf.data(), _zone_conn_info, addr_buf.size());
+        rodsHostAddr_t addr{};
+        auto status = parseHostAddrStr(addr_buf.data(), &addr);
+        if (status < 0) {
+            log_db::info("failed to validate zone connection info [{}]", status);
+            addRErrorMsg(&_ctx.comm()->rError, status, "failed to validate zone connection info");
+            return ERROR(status, "failed to validate zone connection info");
+        }
+        const std::size_t host_len = std::strlen(addr.hostAddr);
+        if (host_len == 0 || (host_len == 1 && addr.hostAddr[0] == ':')) {
+            log_db::info("failed to validate zone connection info due to empty hostname [{}]", _zone_conn_info);
+            addRErrorMsg(&_ctx.comm()->rError,
+                         CAT_HOSTNAME_INVALID,
+                         "failed to validate zone connection info due to empty hostname");
+            return ERROR(CAT_HOSTNAME_INVALID, "failed to validate zone connection info due to empty hostname");
+        }
+        if ((addr.portNum > 65535) || (addr.portNum <= 0)) {
+            log_db::info(
+                "failed to validate zone connection info due to invalid or unspecified port [{}]", _zone_conn_info);
```

Comment:
Is there a reason the `_zone_conn_info` isn't included in the rError message or the `irods::error` message? Same question for the other cases, of course.

Reply:
I vaguely remember the convention being to put this sort of thing in logs, but not in `rError` or `irods::error` messages. If that's not actually the case, I can change it.

Reply:
Seems a good default position to include the thing that didn't work... unless it's a password or other credential/secret I think...